Bug exclude genes

ktrns opened the pull request:
- Fixes issue #88 
- Fixes custom log normalisation to allow exclusion of highly expressed genes
- Generalises assay of raw data into variable so we can later apply the workflow to spatial data #87 
- Uses " for quotation marks
- Uses BE (analyse (i.e. "yse"), normalize (i.e. "ize"), colour (i.e. "our")

ktrns committed:
Added assay parameter to allow the funciton to work with spatial data

ktrns committed:
Debugged function to exclude highly expressed genes. Note that AP replaced apply functions and this still has to be stress-tested.

ktrns committed:
Tiny bugfix

ktrns committed:
Genes are only excluded if they are expressed in less than min_cells in all samples.

ktrns committed:
Merge branch 'master' into bug_excludeGenes

ktrns committed:
Changed quotation marks, and moved assay RNA into a variable so we can later apply the workflow for spatial data.

andpet0101 commented on scrnaseq.Rmd:
```diff
@@ -881,9 +895,10 @@ if (param$norm == "RNA") {
+  # Note: The performance of SCTransform can be improved by using "glmGamPoi" instead of "poisson" as method for initial parameter estimation.
   sc = purrr::map(list_names(sc), function(n) { 
     SCTransform(sc[[n]], 
+                assay=param$assay_raw,
```
Can you add this also to the `SCTransform` in the `integrate`chunk?

ktrns replied:
Will do

ktrns replied:
Done. Question is: I did not do it for `param$norm == "RNA"` ... Since for spatial data, we use `"SCT"`. -Theoretically- it would be possible though ... never tried it. The `param` section atm says: 

```
# Which normalisation should be used for analysis?
#   "RNA" or "SCT" for standard single-cell RNA-seq data
#   "SCT" for spatial data
param$norm = "RNA"
```

ktrns replied:
Fixed for param$norm = "RNA"

andpet0101 commented on scrnaseq.Rmd:
````diff
@@ -1963,6 +1981,10 @@ To gain first insights into potential functions of cells in a cluster, we test f
 We first translate gene symbols of up- and down-regulated genes per cluster into Entrez gene symbols, and then use the "enrichR" R-package to access the "Enrichr" website `r Cite("https://amp.pharm.mssm.edu/Enrichr/", "citep")`. You can choose to test functional enrichment from a wide range of databases:
 ```{r enrichr_databases}
+# Set Enrichr database
+enrichR::setEnrichrSite(param$enrichr_site)
````
Can you deactivate the enrichr messages? Maybe you can tell enrichr to not be verbose?

ktrns replied:
I like the messages, so I know what database it connected to...

andpet0101 commented on scrnaseq.Rmd:
```diff
@@ -232,6 +236,10 @@ param$deg_contrasts = data.frame(condition_column=c("orig.ident", "orig.ident",
 
+# Enrichr site
+# Available options: "Enrichr", "FlyEnrichr", "WormEnrichr", "YeastEnrichr", "FishEnrichr"
+param$enrichr_site = "Enrichr"
```
Can you add this to the `check_enrichr` function so that we always check the correct server?

ktrns replied:
Will do

ktrns replied:
Done

andpet0101 reviewed (changes requested):
Sorry, no PR without changes :-)

Regarding the assay_raw: You also need to change the function `ExportToCerebro`  a bit since there it is still hard-coded "RNA":

```
  cerebroApp::exportFromSeurat(sc,
                               assay=assay,
                               slot="data",
                               file=path,
                               experiment_name="na",
                               organism="na",
                               groups=c("orig.ident", "seurat_clusters"),
                               cell_cycle=c("Phase"),
                               nUMI="nCount_RNA",
                               nGene="nFeature_RNA",
                               add_all_meta_data=TRUE,
                               use_delayed_array=delayed_array,
                               verbose=FALSE)
```

 Maybe just pass `nUMI` and `nGene` as additional arguments...

ktrns commented:
> Sorry, no PR without changes :-)
> 
> Regarding the assay_raw: You also need to change the function `ExportToCerebro` a bit since there it is still hard-coded "RNA":
> 
> ```
>   cerebroApp::exportFromSeurat(sc,
>                                assay=assay,
>                                slot="data",
>                                file=path,
>                                experiment_name="na",
>                                organism="na",
>                                groups=c("orig.ident", "seurat_clusters"),
>                                cell_cycle=c("Phase"),
>                                nUMI="nCount_RNA",
>                                nGene="nFeature_RNA",
>                                add_all_meta_data=TRUE,
>                                use_delayed_array=delayed_array,
>                                verbose=FALSE)
> ```
> 
> Maybe just pass `nUMI` and `nGene` as additional arguments...

Du Fuchs :-), I did set the cerebro chunk to `eval=FALSE` for my spatial data, but you found out that I haven't cared about the Cerebro export! Well done. I changed it, should work now. 

Thanks for the review! I adapted the changes, and will re-run the script now and push the changes. Let's briefly discuss this one question above before merging. 

Best & thanks a lot
Katrin

ktrns committed:
Updated code according to AP's review on the PR

andpet0101 reviewed (approved):
Go ahead

ktrns commented:
- Fixed issue #88
- Fixed custom log normalisation to allow exclusion of highly expressed genes
- Generalises assay of raw data into variable so we can later apply the workflow to spatial data #87
- Uses " for quotation marks
- Uses BE (analyse (i.e. "yse"), normalize (i.e. "ize"), colour (i.e. "our")
